### PR TITLE
#30488: Rewriting 'Next Release' issue labeling for 'QA : Approved' and 'QA : Not Needed'

### LINKED ISSUE #30488: Create a GitHub Action to Automate Labeling for Next Release
### Parent Issue

https://github.com/dotCMS/core/issues/30844

### Task

Create a GitHub Action to automatically add the label Next Release when either the QA Approved or QA Not Needed labels are added to a GitHub issue.

### Proposed Objective

Integrations

### Proposed Priority

Priority 3 - Average

### Acceptance Criteria

- The GitHub Action should trigger when labels are added to an issue or pull request.

- If either the Passed IQA or QA Not Needed label is added, the action should add the Next Release label to the issue.

- The action should ignore cases where the Next Release label is already present.

- The action should log its activity for auditing purposes.

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

1. Trigger Event:
- The action is triggered when a label is added to an issue.
	
2. Check for QA Approved or QA Not Needed Label:
- The first step uses the github-script action to inspect the labels on the issue.
- It checks if either the QA Approved or QA Not Needed label is present and verifies that Next Release is not already added.
- If both conditions are met, the result is set to true, allowing the next step to run.
	
3. Add Next Release Label:
- If the previous check passed, this step uses the action-add-labels action to add the Next Release label to the issue.
	
4. Log Action:
- Logs a message indicating that the Next Release label was added due to the presence of QA Approved or QA Not Needed.

### Quality Assurance Notes & Workarounds

- GitHub Token: The ${{ secrets.GITHUB_TOKEN }} is used for authentication, allowing the action to add labels to issues.

- Logging: This workflow logs a message each time it successfully adds the Next Release label, which is useful for tracking and auditing purposes.

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by victoralfaro-dotcms**
The internal workflow that takes care of the actual labeling is more flexible and generic so all that was hardcoded is parametrized now.

This PR fixes: #30488

### CODE REVIEW

**Review by jcastro-dotcms — APPROVED** [review]

File: .github/workflows/issue_comp_label-conditional-labeling.yml [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+      evaluating_labels:
+        description: 'Comma delimited list of labels to evaluate against'
+        type: string
+        required: true
+      evaluated_labels:
+        description: 'Comma delimited list of labels to be evaluated'
+        type: string
+        required: true
+      ignore_issue_labels:
+        description: 'Comma delimited list list of issue labels to ignore'
```
Comment (dcolina) [inline .github/workflows/issue_comp_label-conditional-labeling.yml:19]: There is a typo on description

File: .github/workflows/issue_comp_label-conditional-labeling.yml [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+            if (!issue) {
+              core.warning('Issue [${{ inputs.issue_number }}] not found');
+              process.exit(0);
+            }
+            
+            console.log(`Issue: ${JSON.stringify(issue, null, 2)}`);
+            const issueNumber = issue.number;
+            const foundLabel = ignoreIssueLabels.find(lbl => issue.labels.includes(lbl));
+            if (foundLabel) {
+              core.warning(`Issue does have label in [${ignoreIssueLabels.joint(', ')}] ignore labels group, aborting...`);
```
Comment (dcolina) [inline .github/workflows/issue_comp_label-conditional-labeling.yml:129]: There is a typo 'joint'. I think it must be 'join' instead.

File: .github/workflows/issue_comp_label-conditional-labeling.yml [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+            const evaluatingLabels = '${{ inputs.evaluating_labels }}'.split(',').map(lbl => lbl.trim());
+            
+            let issueNumber;
+            const issue = context.payload.issue;
+            if (!issue && '${{ inputs.issue_number }}'.trim() === '') {
+              core.warning('Issue number is not provided');
+              process.exit(0);
+            }
+            
+            if (!'${{ inputs.evaluated_labels }}') {
```
Comment (dcolina) [inline .github/workflows/issue_comp_label-conditional-labeling.yml:80]: This conditional might always be false if the string is not empty.

File: .github/workflows/issue_comp_label-conditional-labeling.yml [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+        description: 'Comma delimited list of labels to be evaluated'
+        type: string
+        required: true
+      ignore_issue_labels:
+        description: 'Comma delimited list list of issue labels to ignore'
+        type: string
+        required: false
+        default: ''
+      new_labels:
+        description: 'Comma delimited list list of new labels to be added to issue'
```
[the remark this answers is not in the document]
Comment (victoralfaro-dotcms) [reply .github/workflows/issue_comp_label-conditional-labeling.yml:24]: Fixed

File: .github/workflows/issue_comp_label-conditional-labeling.yml (hunk 1 above)
Comment (victoralfaro-dotcms) [reply .github/workflows/issue_comp_label-conditional-labeling.yml:19]: Fixed

**Review by dcolina — APPROVED** [review]
Looks good 👍🏽

**Pull request merged**